test/src/api/commands/element/testSetPasswordCheckReports.js

```diff
@@ -0,0 +1,80 @@
+const assert = require('assert');
+const path = require('path');
+const {Key} = require('selenium-webdriver');
+const common = require('../../../../common.js');
+const {settings} = common;
+const NightwatchClient = common.require('index.js');
+const MockServer  = require('../../../../lib/mockserver.js');
+const CommandGlobals = require('../../../../lib/globals/commands.js');
+
+describe('setPassword check', function() {
+
+  before(function(done) {
+    this.server = MockServer.init();
+    this.server.on('listening', () => done());
+  });
+
+  after(function(done) {
+    this.server.close(function() {
+      done();
+    });
+  });
+
+  it('client.setPassword() value redacted in rawHttpOutput', async function() {
+
+    MockServer.addMock({
+      url: '/wd/hub/session/1352110219202/element/0/value',
+      method: 'POST',
+      postdata: {
+        text: Key.NULL + 'password',
+        value: [Key.NULL, 'p', 'a', 's', 's', 'w', 'o', 'r', 'd']
+      },
+      response: {
+        sessionId: '1352110219202',
+        status: 0
+      }
+    });
```

Comment:
We need to make sure that these mocks are actually encountered. For that, we can define a `sendKeysRedactedMockCalled` variable at the top, and then add a `onRequest` method in the object passed to `.addMock()` command (which will run each time the mock is hit) and in there set the defined `sendKeysRedactedMockCalled` variable to `true`.

At last, assert the variable `sendKeysRedactedMockCalled` for truth value at the end of the end (after `await NightwatchClient...` call).

Reply:
added [9ad5d68](https://github.com/nightwatchjs/nightwatch/pull/4066/commits/9ad5d68ae6a1aa07a9b487d389e95de41976f225)